File: task2/main.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

int minRefuels(int N, int V, const vector<int>& R) {
    for (int dist : R) {
        if (dist > V) return 0;
    }

    int fuel = V;
    int refuels = 0;

    for (int i = 0; i < N - 1; i++) {
        if (fuel < R[i]) return 0;

        fuel -= R[i];

        if (i < N - 2 && fuel < R[i + 1]) {
            refuels++;
            fuel = V;
        }
    }

    return refuels;
}

int main() {
    ifstream fin("input.txt");
    ofstream fout("output.txt");

    int N, V;
    vector<int> R;

    if (fin.is_open()) {
        fin >> N >> V;
        R.resize(N - 1);
        for (int i = 0; i < N - 1; i++) {
            fin >> R[i];
        }
    } else {
        cout << "Введите количество планет N: ";
        cin >> N;
        cout << "Введите емкость бака V: ";
        cin >> V;

        R.resize(N - 1);
        cout << "Введите расстояния между соседними планетами: ";
        for (int i = 0; i < N - 1; i++) {
            cin >> R[i];
        }
    }

    int result = minRefuels(N, V, R);

    cout << "Результат: " << result << endl;
    fout << result << endl;

    return 0;
}

File: task2/main.rs
use std::fs::{read_to_string, File};
use std::io::{self, Write};

fn min_refuels(n: usize, v: i32, r: &Vec<i32>) -> i32 {
    for &dist in r {
        if dist > v {
            return 0;
        }
    }

    let mut fuel = v;
    let mut refuels = 0;

    for i in 0..(n - 1) {
        if fuel < r[i] {
            return 0;
        }

        fuel -= r[i];

        if i < n - 2 && fuel < r[i + 1] {
            refuels += 1;
            fuel = v;
        }
    }

    refuels
}

fn main() {
    let mut n: usize = 0;
    let mut v: i32 = 0;
    let mut r: Vec<i32> = Vec::new();

    if let Ok(content) = read_to_string("input.txt") {
        let nums: Vec<i32> = content
            .split_whitespace()
            .map(|x| x.parse::<i32>().unwrap())
            .collect();

        n = nums[0] as usize;
        v = nums[1];
        r = nums[2..].to_vec();
    } else {
        let mut input = String::new();

        print!("Введите количество планет N: ");
        io::stdout().flush().unwrap();
        io::stdin().read_line(&mut input).unwrap();
        n = input.trim().parse::<usize>().unwrap();

        input.clear();
        print!("Введите емкость бака V: ");
        io::stdout().flush().unwrap();
        io::stdin().read_line(&mut input).unwrap();
        v = input.trim().parse::<i32>().unwrap();

        input.clear();
        print!("Введите расстояния между соседними планетами: ");
        io::stdout().flush().unwrap();
        io::stdin().read_line(&mut input).unwrap();
        r = input
            .split_whitespace()
            .map(|x| x.parse::<i32>().unwrap())
            .collect();
    }

    let result = min_refuels(n, v, &r);

    println!("Результат: {}", result);

    let mut file = File::create("output.txt").unwrap();
    writeln!(file, "{}", result).unwrap();
}
